### PR TITLE
Fix build wamrc for xtensa failure

### PR DESCRIPTION
**Pull request by wenyongh**

### CODE REVIEW

**Comment by yamt** [comment]
lgtm. why is this still WIP?

a question: do we need to keep old llvm support?
the xtensa llvm repo seems to have releases based on llvm 13 and even 14.

**Comment by wenyongh** [comment]
> lgtm. why is this still WIP?
> 

Yes, it looks good, but not very sure, I need to test more.

> a question: do we need to keep old llvm support? the xtensa llvm repo seems to have releases based on llvm 13 and even 14.

Had better support llvm-11 and llvm-12, some developers may still use them. OK, seems that we should upgrade the script to use xtensa llvm 13.
For llvm-14, we may upgrade WAMR to support it in some time, besides with the toolchain emsdk and wasi-sdk, to use the latest version. Currently we are planning to create a new release, we may upgrade them after that.

File: core/iwasm/compilation/aot_llvm_extra.cpp [hunk 1]
```diff
@@ -478,3 +479,4 @@ aot_apply_llvm_new_pass_manager(AOTCompContext *comp_ctx)
     }
 #endif
 }
+#endif
```
Comment (xujuntwt95329) [inline core/iwasm/compilation/aot_llvm_extra.cpp:482]: Had better add a comment:
```suggestion
#endif /* end of LLVM_VERSION_MAJOR >= 12 */
```

**Review by xujuntwt95329 — COMMENTED** [review]
LGTM

File: core/iwasm/compilation/aot_llvm_extra.cpp (hunk 1 above)
Comment (wenyongh) [reply core/iwasm/compilation/aot_llvm_extra.cpp]: Done

**Pull request merged**